mozetl/churn/churn.py

```diff
@@ -79,12 +79,50 @@
 DEFAULT_DATE = '2000-01-01'  # The default date used for cleaning columns
 
 
-def extract_subset(main_summary, start_ds, period, slack, is_sampled):
+def clean_new_profile(new_profile):
+    """Create a `main_summary` compatible dataset from the new profile ping.
+
+    :param new_profile:  dataframe conforming to the new profile ping
+    :returns:            subset of `main_summary` used in churn
+    """
+
+    iso8601_tz_format = "yyyy-MM-dd'T'HH:mm:ss.S'+00:00'"
+
+    select_expr = {
+        "app_version": "environment.build.version",
+        "attribution": "environment.settings.attribution",
+        "client_id": None,
+        "country": "metadata.geo_country",
+        "default_search_engine": "environment.settings.default_search_engine",
+        "distribution_id": "environment.partner.distribution_id",
+        "locale": "environment.settings.locale",
+        "normalized_channel": "metadata.normalized_channel",
+        "profile_creation_date": "environment.profile.creation_date",
+        "subsession_length": F.lit(None).cast('long'),
+        # The subsession_start_date is the profile_creation_date
+        "subsession_start_date": F.from_unixtime(
+            F.col("metadata.creation_timestamp") / 10 ** 9, iso8601_tz_format),
+        # no access to `WEAVE_CONFIGURED`, `WEAVE_DEVICE_COUNT_*` histograms in the new_profile
+        "sync_configured": F.lit(None).cast("boolean"),
+        "sync_count_desktop": F.lit(None).cast("int"),
+        "sync_count_mobile": F.lit(None).cast("int"),
+        "timestamp": F.col("metadata.timestamp"),
+        "scalar_parent_browser_engagement_total_uri_count": F.lit(None).cast("int"),
+        "scalar_parent_browser_engagement_unique_domains_count": F.lit(None).cast("int"),
+        "sample_id": F.expr("crc32(encode(client_id, 'UTF-8')) % 100"),
```

Comment:
Please add a test case that ensures the `sample_id` computation matches with values from main_summary for a few client_ids. If you want you can use mine:
`client_id = 'c4582ba1-79fc-1f47-ae2a-671118dccd8b', sample_id = 4`